### ISSUE TITLE
Problem with components using `sustain`

### ISSUE THREAD

**Issue by greyhwndz** [runspired/flexi#49]
I have a new ember app with the following versions:
Ember      : 2.4.3
Ember Data : 2.4.3+a3e2521f85
jQuery     : 2.2.2

_addons:_
**ember-cli-mirage**
**flexi v1.1.4**

I have followed the example in:
https://github.com/runspired/flexi#example-layout

I am having the following error:
```
sustain.js:224 Uncaught TypeError: Cannot read property 'get' of undefined
setupComponent @ sustain.js:224
```

What follows is my code:

**ember-cli-build.js**
```
var EmberApp = require('ember-cli/lib/broccoli/ember-app');
var shim = require('flexi/lib/pod-templates-shim');

shim(EmberApp);
```

**app/router.js**
```
Router.map(function() {
	this.route('emails', function(){
		this.route('index', { path: '/' });
		this.route('single', { path: '/:id' });
	});
});
```

**app/routes/emails/route.js**
```
export default Ember.Route.extend({
	model() {
		return Ember.RSVP.hash({
			emails: this.store.findAll('email')
		});
	}
});
```

**app/routes/emails/-layouts/desktop.hbs**
```
<screen>
  <page>
    <hbox>
      <vbox md="4">
      	{{sustain 'emails/components/email-list' model}}
      </vbox>
    </hbox>
    <vbox md="8">
        {{liquid-outlet "main"}}
    </vbox>
  </page>
</screen>
```

**app/routes/emails/components/email-list.hbs**
```
<ul>
{{#each model.emails as |email|}}
	<li>{{#link-to 'emails.single' email}}{{email.title}}{{/link-to}}</li>
{{/each}}
</ul>
```

**app/routes/emails/index/route.js**
```
export default Ember.Route.extend({
	model() {
		return this.modelFor('emails');
	}
});
```

**app/routes/emails/index/-layouts/desktop.hbs**
```
<p>
Please select an email on the left to see its details.
</p>
```

**app/routes/emails/single/route.js**
```
export default Ember.Route.extend({
	model(params) {
		return this.store.findRecord('email', params.id);
	}
});
```

**app/routes/emails/single/-layouts/desktop.hbs**
```
Email Title: {{model.title}}
```

NOTE: I have the usual endpoints in my `mirage/config.js` 
and some `fixtures` for `/emails` that works.

If I swap the following code:
```
{{#emails/components/email-list model}}
```
in lieu of:
```
{{sustain 'emails/components/email-list' model}}
```
in **`app/routes/emails/-layouts/desktop.hbs`**

I get the following error in the browser:
```
Build error

flexi-app/routes/emails/template.hbs (in /Users/greyhwndz/flexi-app/tmp/template_compiler-input_base_path-3rsVJARa.tmp/0)

The Broccoli Plugin: [TemplateCompiler] failed with:
Error: emails/components/email-list doesn't match with - 18:11
    at Object.prepareBlock (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:16842:13)
    at Object.anonymous (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:16917:37)
    at Object.parse (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:17185:52)
    at Object.parse (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:16739:73)
    at Object.preprocess (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:18483:87)
    at compileSpec (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:12971:37)
    at Object.exports.default [as precompile] (/Users/greyhwndz/flexi-app/bower_components/ember/ember-template-compiler.js:12915:12)
    at Object.module.exports.template (/Users/greyhwndz/flexi-app/node_modules/ember-cli-htmlbars/utils.js:29:40)
    at TemplateCompiler.processString (/Users/greyhwndz/flexi-app/node_modules/ember-cli-htmlbars/index.js:68:36)
    at Promise.then.result.output (/Users/greyhwndz/flexi-app/node_modules/broccoli-persistent-filter/lib/strategies/persistent.js:41:23)

The broccoli plugin was instantiated at: 
    at TemplateCompiler.Plugin (/Users/greyhwndz/flexi-app/node_modules/broccoli-plugin/index.js:7:31)
    at TemplateCompiler.Filter [as constructor] (/Users/greyhwndz/flexi-app/node_modules/broccoli-persistent-filter/index.js:38:10)
    at new TemplateCompiler (/Users/greyhwndz/flexi-app/node_modules/ember-cli-htmlbars/index.js:20:10)
    at TemplateCompiler (/Users/greyhwndz/flexi-app/node_modules/ember-cli-htmlbars/index.js:12:12)
    at Object.registry.add.toTree (/Users/greyhwndz/flexi-app/node_modules/ember-cli-htmlbars/ember-addon-main.js:31:16)
    at /Users/greyhwndz/flexi-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:184:26
    at Array.forEach (native)
    at processPlugins (/Users/greyhwndz/flexi-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:182:11)
    at module.exports.preprocessTemplates (/Users/greyhwndz/flexi-app/node_modules/ember-cli-preprocess-registry/preprocessors.js:165:10)
    at EmberApp._processedTemplatesTree (/Users/greyhwndz/flexi-app/node_modules/ember-cli/lib/broccoli/ember-app.js:802:48)
```

**Comment by greyhwndz** [comment]
Turns out that the podified component needs to be updated to be in the correct podified format i.e.:
`app/routes/emails/components/email-list/template.hbs` &
`app/routes/emails/components/email-list/component.js` instead of just
`app/routes/emails/components/email-list.hbs`

created a PR for this

**Comment by runspired** [comment]
Resolved by #50
